### PR TITLE
support redis-rb 4.0.0

### PR DESCRIPTION
**Pull request by ota42y**
The redis-rb 4.0.0 remove [] and []= method alias.
So we must use get/set method.

And `client` method was broken changed.
https://github.com/redis/redis-rb/commit/31385074b6bbeef7e1f9849b0b1149b9ef870e2d

We must use the `_client` method for same behavier.
https://github.com/redis/redis-rb/commit/c239abb43c2dce99468bf94652a3c48b31a1041a

But if we remove `client` method, it's broken changes, so I alias `client` method to `_client`.
The user can be used redis-naespace 1.x with both redis-3.x and redis-4.x and can be gradually update.

### CODE REVIEW

**Comment by ota42y** [comment]
@fw42 
This feature doesn't changed the original behavior, I think it's no problem.
And this feature enable using redis-rb v4.0.0, so this is very useful for redis-namespace user.

File: lib/redis/namespace.rb [hunk 1]
```diff
@@ -241,6 +241,7 @@ def initialize(namespace, options = {})
       @deprecations = !!options.fetch(:deprecations) do
                         ENV['REDIS_NAMESPACE_DEPRECATIONS']
                       end
+      @is_client_respond = @redis.respond_to?(:_client)
```
Comment (fw42) [inline lib/redis/namespace.rb]: I'd call this something like `@has_new_client_method` or `@responds_to_new_client_method`... or maybe just `@new_client_method`.

File: lib/redis/namespace.rb [hunk 2]
```diff
@@ -252,7 +253,13 @@ def warning?
     end
 
     def client
-      @redis.client
+      warn("The `client` method broken changed by redis-rb 4.0.0, so we should use `_client` method." +
+               "`client` method in redis-namespace will remove 2.0 (at #{call_site})") if @is_client_respond
```
Comment (fw42) [inline lib/redis/namespace.rb]: I'd say something like
> The `client` method is deprecated as of redis-rb 4.0.0, please use the new `_client` method instead. Support for the old method will be removed in redis-namespace 2.0.

**Comment by ota42y** [comment]
@fw42 

Fixed!
Thank you for advice of more correct English expression

**Comment by ota42y** [comment]
Thanks!
And when will we release the version containing this change?

**Comment by jeremy** [comment]
resque-scheduler support for redis-rb 4.0 (https://github.com/resque/resque-scheduler/pull/620) is pending a gem release it can depend on.

File: redis-namespace.gemspec [hunk 3]
```diff
@@ -18,7 +18,7 @@ Gem::Specification.new do |s|
   s.files            += Dir.glob("test/**/*")
   s.files            += Dir.glob("spec/**/*")
 
-  s.add_dependency    "redis", "~> 3.0", ">= 3.0.4"
+  s.add_dependency    "redis", ">= 3.0.4"
```
Comment (D-system) [inline redis-namespace.gemspec]: Might be better to specify a range of compatibility:
`s.add_dependency    "redis", ">= 3.0.4", "< 5.0"`

**Comment by manuelmeurer** [comment]
Could you release a new gem version with Redis v4 support? 😸  Thanks!

**Comment by ota42y** [comment]
@fw42
Hi! could you release new version?
I want to use redis 4 😃

File: redis-namespace.gemspec (hunk 3 above)
Comment (PikachuEXE) [inline redis-namespace.gemspec]: Not `<5`?

**Comment by jeremy** [comment]
[1.6.0](https://rubygems.org/gems/redis-namespace/versions/1.6.0) was released

**Pull request merged**